test/integration/apiserver/flowcontrol/concurrency_util_test.go

```diff
@@ -0,0 +1,496 @@
+
+	queueLength := 50
+	concurrencyShares := 100
+
+	priorityLevelNoxu1, _, err := createPriorityLevelAndBindingFlowSchemaForUser(
+		loopbackClient, "noxu1", concurrencyShares, queueLength)
+	if err != nil {
+		t.Error(err)
+	}
+	priorityLevelNoxu2, _, err := createPriorityLevelAndBindingFlowSchemaForUser(
+		loopbackClient, "noxu2", concurrencyShares, queueLength)
+	if err != nil {
+		t.Error(err)
+	}
+	availableSeats, err := getAvailableSeatsOfPriorityLevel(loopbackClient)
+	if err != nil {
+		t.Error(err)
+	}
+
+	t.Logf("noxu1 priority level concurrency limit: %v", availableSeats[priorityLevelNoxu1.Name])
+	t.Logf("noxu2 priority level concurrency limit: %v", availableSeats[priorityLevelNoxu2.Name])
+	if (availableSeats[priorityLevelNoxu1.Name] <= 4) || (availableSeats[priorityLevelNoxu2.Name] <= 4) {
+		t.Errorf("The number of available seats for test client priority levels are too small: (%v, %v). Expecting a number > 4",
+			availableSeats[priorityLevelNoxu1.Name], availableSeats[priorityLevelNoxu2.Name])
+	}
+
+	stopCh := make(chan struct{})
+	wg := sync.WaitGroup{}
+
+	// "elephant"
+	noxu1NumGoroutines := 5 + queueLength
+	var noxu1ClientRequestLatencySum float64
+	var noxu1ClientRequestLatencySumSq float64
+	var noxu1ClientRequestLatencyCount int32
+	var noxu1Mutex sync.Mutex
+	streamRequestsWithIndex(noxu1NumGoroutines, func(idx int) {
+		start := time.Now()
+		_, err := noxu1Client.CoreV1().Namespaces().Get(context.Background(), "default", metav1.GetOptions{})
+		duration := time.Since(start).Seconds()
+		noxu1Mutex.Lock()
+		noxu1ClientRequestLatencyCount += 1
+		noxu1ClientRequestLatencySum += duration
+		noxu1ClientRequestLatencySumSq += duration * duration
+		noxu1Mutex.Unlock()
+		if err != nil {
+			t.Error(err)
+		}
+	}, &wg, stopCh)
+	// "mouse"
+	noxu2NumGoroutines := 3
+	var noxu2ClientRequestLatencySum float64
+	var noxu2ClientRequestLatencySumSq float64
+	var noxu2ClientRequestLatencyCount int32
+	var noxu2Mutex sync.Mutex
+	streamRequestsWithIndex(noxu2NumGoroutines, func(idx int) {
+		start := time.Now()
+		_, err := noxu2Client.CoreV1().Namespaces().Get(context.Background(), "default", metav1.GetOptions{})
+		duration := time.Since(start).Seconds()
+		noxu2Mutex.Lock()
+		noxu2ClientRequestLatencyCount += 1
+		noxu2ClientRequestLatencySum += duration
+		noxu2ClientRequestLatencySumSq += duration * duration
+		noxu2Mutex.Unlock()
+		if err != nil {
+			t.Error(err)
+		}
+	}, &wg, stopCh)
+
+	// Warm up
+	time.Sleep(testWarmUpTime)
+
+	// Reset counters
+	noxu1Mutex.Lock()
+	noxu1ClientRequestLatencyCount = 0
+	noxu1ClientRequestLatencySum = 0
+	noxu1ClientRequestLatencySumSq = 0
+	noxu1Mutex.Unlock()
+	noxu2Mutex.Lock()
+	noxu2ClientRequestLatencyCount = 0
+	noxu2ClientRequestLatencySum = 0
+	noxu2ClientRequestLatencySumSq = 0
+	noxu2Mutex.Unlock()
+	earlierRequestExecutionSecondsSum, earlierRequestExecutionSecondsCount, earlierPLSeatUtilSum, earlierPLSeatUtilCount, err := getRequestExecutionMetrics(loopbackClient)
```

Comment:
These names are super long, that combined with repetition makes this code very hard to read.

I would declare a struct type and put these things inside it. I'd further make a sum/count pair struct.

Rename:
* earlier -> t0
* later -> t1
* RequestExecutionSecondsSum -> reqExecution.Sum
* RequestExecutionSecondsCount -> reqExecution.Count
* PLSeatUtilSum -> seat.Sum
* PLSeatUtilCount -> seat.Count

So the below math can say things like t0.reqExecution.Sum.

Put comments by the variable names that expand the terse names.

Reply:
Code is refactored in the new version